rand_core/src/lib.rs

```diff
@@ -44,6 +44,8 @@
 
 #[cfg(feature="std")] extern crate core;
 #[cfg(all(feature = "alloc", not(feature="std")))] extern crate alloc;
+#[cfg(feature="serde-1")] extern crate serde;
```

Comment:
I have a weak preference for `serde1` as the name of the Serde feature instead of `serde-1`, to be consistent with crates that enable optional Serde support through an optional dependency on the [`serde1`](https://crates.io/crates/serde1) shim.

Reply:
From what I see `serde-1` seems to be used by a couple of crates to add the optional dependency. `serde1` does not seem to have any reverse dependencies?

I don't like `serde-1`, and `serde1` also does not seem ideal compared to just `serde`.
For `rand_core`, we could name the feature just `serde` with:
`serde = { version = "1", optional = true, features = ["derive"] }`.

For `rand` I see the serde feature as temporary, until we split of the PRNGs into a separate crate. There seems to be some movement in https://github.com/rust-lang/cargo/issues/1286.

@dhardy What do you think? Does it make sense to use `serde` as the feature name in `rand_core`, and at some point in the future both crates can use that name?

Reply:
Well if `serde` never has another breaking release we don't need to append any number, but since it had quite a few before 1.0 appending a number seemed prudent. If @dtolnay prefers `serde1` that is fine, except that it's a breaking change (we added the `serde-1` feature for the Rand 0.4 release).

Reply:
> except that it's a breaking change (we added the `serde-1` feature for the Rand 0.4 release).

I don't think we did? https://github.com/rust-lang-nursery/rand/blob/0.4/Cargo.toml

Reply:
Okay, then we aren't constrained and can use `serde1` as suggested.